src/cargo/core/resolver/features.rs

```diff
@@ -321,21 +322,14 @@ impl ResolvedFeatures {
             .expect("activated_features for invalid package")
```

Comment:
I feel like we are not on the same page, @ehuss 😆 

> > Maybe I miss some contexts, but it seems that we can switch to unwrap_or_default nowadays even if we stop inserting empty set from now on.
> 
> I'm not sure I follow. With this change, it must insert an empty set, so skipping that step is no longer an option?


`activated_features_int` returns a `Result` so `unwrap_or_default` can easily recover from it with a default. There is no need of the presence of an empty set, no?

https://github.com/rust-lang/cargo/blob/1945b44853c87bc8a600ced428e1292a5350e814/src/cargo/core/resolver/features.rs#L352-L363

Reply:
Ah, I was conflating "switch to unwrap_or_default" and "remove inserting the empty set", as I was thinking of them as a single action that would be done together. I suppose it could switch to unwrap_or_default, but I like the validation that `expect` has been giving.